### PR TITLE
Expose that the python API is Python3.5+ only

### PR DESCRIPTION
**Pull request by TimZaman**
Adding these classifiers (many are the same as e.g. TensorFlow) makes sure pip/pypi index pick these up as being 3.5+ only, so that developers on Python 2 are not suprised when it fails on Python 3 syntax (e.g. typehints are only py 3.5+).

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/616)

### CODE REVIEW

**Comment by yebrahim** [comment]
I think we're actually Python 3.6+ only. @qimingj @gaoning777 @Ark-kun

**Comment by TimZaman** [comment]
Thanks @yebrahim , reflected that in change. Tested 3.7 to work too.

**Comment by yebrahim** [comment]
Yup, >= 3.6. I'll let @qimingj or @gaoning777 take a look first though. Thanks @TimZaman.

File: sdk/python/setup.py [hunk 1]
```diff
@@ -41,6 +41,20 @@
       'kfp_run.api',
       'kfp_run.models',
     ],
+    classifiers=[
+      'Intended Audience :: Developers',
+      'Intended Audience :: Education',
+      'Intended Audience :: Science/Research',
+      'License :: OSI Approved :: Apache Software License',
+      'Programming Language :: Python :: 3',
+      'Programming Language :: Python :: 3.6',
```
Comment (qimingj) [inline sdk/python/setup.py]: I think the SDK is good on py 3.5 too. Add 3.5?

Comment (TimZaman) [reply sdk/python/setup.py]: Done

**Comment by TimZaman** [comment]
@qimingj done, should be g2g

**Comment by Ark-kun** [comment]
> I think we're actually Python 3.6+ only.

I thought we were 3.5+. 3.6+ is much better for me (variable typing).

**Comment by Ark-kun** [comment]
We should probably just use `python-requires` in `setup.py`. See https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires

**Comment by TimZaman** [comment]
Added `python_requires`.

**Comment by TimZaman** [comment]
I think this one is g2g

**Pull request merged**